### PR TITLE
Sahaj footer pt 90-106

### PR DESCRIPTION
**Pull request by sahajisy**
```
@FooterPage
Feature: Validate footer links on Footer Sections

#Testcase no: PT-090 to 95
Scenario: Validate Footer Links on Top Page

Then Check footer links on Top Page

#Testcase no: PT-122
Scenario: Check Links on Terms and Conditions Page

Then Check links in Terms of Use Page

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Account and Services Page 
Then Check footer links on Login Page

#Testcase no: PT-096 to PT-101
Scenario: Validate Footer on User Top Page 

Given Login to Platform Portal
Then Check footer links on top page after login

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen One

Given Login to Platform Portal
Then Check Footer Links on First Order Screen

#Testcase no: PT-102 to PT-105
Scenario: Validate Footer on Order Screen Two and Three

Given Login to Platform Portal
Then Check Footer Links on Second Order Screen
Then Check Footer Links on Third Order `Screen```
```

### CODE REVIEW

**Comment by csonparote** [comment]
Please create exclusive tag name for scenario. will be useful for future tagging.
Please resolve error as per call

**Comment by sahajisy** [comment]
@csonparote I have added the Tag for Feature file also did the necessary changes for code to execute.

**Comment by sahajisy** [comment]
@csonparote Handled the close tab issue on footer links for external pages.

**Comment by csonparote** [comment]
@sahaj-hirameki Please resolve conflicting files.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Please create a new code block for authorship. dont merge into others

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Remove Thread.sleep mention from code.
Use wait from CommonFunctions for the same

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Please refactor/rename method in verb-noun format.
It is difficult to know what this method does

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Rename to verb-noun format

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Plz mention type of element in name itself.
ex: casecTermsLink etc

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Rename to verb-noun format
Cant determine what this method does

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Rename to verb-noun format

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature]: Name is exclusive which is good.
Adding feature file name to tag would be better

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature]: Please break open browser into seperate step
Ex: Given Navigate to URL <browser>
When User clicks on footer links

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature]: Ex: 
Then validate links can be opened
And close browser

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Footer.java]: Replace with public WebDriver driver = ObjectHelper.driver;
Needed to retain browser instance

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Footer.java]: Rename to verb-non format
Cant determine what this method does

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Footer.java]: Rename to verb-non format
Cant determine what this method does

**Review by csonparote — CHANGES_REQUESTED** [review]
Please do the needful

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Done the following change

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Removed Thread.sleep function

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Done changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Done changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Will change some Web element names

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: Done changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Please add your seperate authorship block for code changes that u have done

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: Used And feature.

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java]: Done changes as requested.

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java]: Done changes as requested will confirm it again.

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java]: Done changes as requested.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage.java
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage.java]: Looks Good. thanks

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: yup. We will edit it after merging to work on new runner

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java]: needs few edit for new runner. will discuss after code merge. Its good for now

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java]: Thanks

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: As per above comment, I cant determine what are you validating in Then step. Please add new step

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Terms and Conditions Pageobject is being handled by Sanae san.

**Comment by sahajisy** [comment]
Addded Some new features and step defs for Footer links on Top Page, Top Page after Login and Login Page.
Can you check it and try to execute it on you side?

**Comment by sahajisy** [comment]
I am not able to handle authentication popup for jyuku pita, login and registration footer links. How can I handle it?

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Plz coordinate with Sanae san for the same.
Ensure that it is working fine in your code

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: Updated Feature file.

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature]: Your validation starts here So this statement can be prefixed with Then.

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature:22]: Multiple then statements not needed. You can prefix with And after one then statement.

**Comment by sahajisy** [comment]
@csonparote san, 
Updated Footerpage file name to FooterPage_PT90_106 to avoid conflicts in Git.
Added FooterAssert sheet in NewTestData.xlsx.
Handled authentication popup for create account page title assertion.

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: I have updated Terms and Conditions Code.

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Footer.feature]: Updated Feature file.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace abs xpath with relative ones

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Please remove commented code, if you are never planning to use it

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: Plz replace absolute xpath with relative ones wherever applicable

File: platform/src/test/java/platformfeatures/PF_Footer.feature
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Footer.feature:2]: Assign unique tag to feature file

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Footer.java:77]: Remove commented code if not needed

**Review by csonparote — CHANGES_REQUESTED** [review]
Please do the needful

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: I checked the xpaths for terms and conditions links but they have same class so It when I added those classes xpaths, it was not identifying. So had to take unique class from from webpage and had to add xpath from then on.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: I checked the xpaths for Footer links but they have same class so It when I added those classes xpaths, it was not identifying. So had to take unique class from from webpage and had to add xpath from then on.

File: platform/src/test/java/platformstepdefinition/PF_Footer.java
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Footer.java:77]: Done the requested changes.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Done the requested changes.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: In case of design change, all these links will not be found as it is absolute path. Plz replace to relative xpath

**Review by csonparote — CHANGES_REQUESTED** [review]
Plz do needful

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: @csonparote san, I have updated the xpaths.

File: platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/TermsConditionsFooter.java]: @csonparote san, I have updated the xpaths.

**Review by csonparote — APPROVED** [review]
Plz merge main into ur branch and check

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: xpath not updated in this file

**Review by csonparote — CHANGES_REQUESTED** [review]
Please change xpath in platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace this abs path

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace this abs path

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace with xpath without contains(text())

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: PLz replace if possible. img tag is not advisable

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace with xpath without contains(text())

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace with xpath without contains(text())

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Plz replace img tag with alternate xpath

**Review by csonparote — CHANGES_REQUESTED** [review]
Plz make changes

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: @csonparote san, I have updated the xpaths.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: @csonparote san, I have updated the xpaths.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated Webelement name
Following Webelement is used for visibility.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated changes as requested

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated Webelement name
Following Webelement is used for visibility.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated xpaths as requested.

File: platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Footer/FooterPage_PT90_106.java]: Updated xpaths as requested.

**Review by csonparote — APPROVED** [review]
Plz merge main into this branch and check

**Pull request merged**
